### PR TITLE
added MUI and react-icons

### PR DESCRIPTION
**Pull request by samuelgodinho02112020**
The first task of installing Material-UI (MUI) into the project is completed. Moving forward, I've carefully evaluated two icon libraries: React-Icons and Heroicons. After thorough consideration, I've decided to proceed with React-Icons due to its ease of installation and preferred usage in React projects. It's a free library with seamless integration and boasts an impressive 6.3K stars on GitHub.

As for Heroicons, I've opted not to utilize them primarily because they are coupled with TailwindCSS, a framework I'm not yet familiar with. Additionally, while Material-UI also offers its own set of icons, I believe React-Icons surpasses its functionality and usability for our specific project needs.

### CODE REVIEW

File: frontend/package.json [hunk 1]
```diff
@@ -9,15 +9,21 @@
     "lint": "next lint"
   },
   "dependencies": {
+    "@emotion/cache": "^11.11.0",
+    "@emotion/react": "^11.11.4",
+    "@emotion/styled": "^11.11.0",
```
Comment (castorgodinho) [inline frontend/package.json:14]: What is `@emotion` package for?

File: frontend/package.json [hunk 2]
```diff
@@ -9,15 +9,21 @@
...
+    "@mui/material-nextjs": "^5.15.11",
+    "@types/jest": "^29.5.12",
```
Comment (castorgodinho) [inline frontend/package.json:16]: Do we need `@types/jest`?

File: frontend/package.json (hunk 1 above)
Comment (samuelgodinho02112020) [reply frontend/package.json:14]: Emotion is a JavaScript library that allows you to write CSS styles directly in your JavaScript code. This approach offers benefits like easy style composition and a smoother developer experience. 
 i have used  "@emotion/cache": "^11.11.0",  becusae 
In simple terms, @emotion/cache is like a memory bank for styles in Emotion. It stores styles that have been used before, so when your website or app loads, it doesn't have to recreate those styles from scratch every time. This makes your website faster because it doesn't have to do unnecessary work, and it helps prevent duplicate styles from cluttering up your code. Overall, it's a way to make your styling process more efficient and your website faster.
 
"@emotion/react": "^11.11.4",
In simple terms, @emotion/react is a package that allows you to style your React components in a more organized and efficient way. It provides tools that make it easier to write and manage CSS styles directly within your React code. This means you can make your components look exactly how you want them to without needing separate CSS files.

"@emotion/styled": "^11.11.0",
@emotion/styled is a package that lets you create styled components in your React applications using Emotion. It's like a special tool that makes it easy to style your components by writing CSS inside your JavaScript or TypeScript files. This way, you can keep your styling code close to your component code, making it easier to manage and understand.

Comment (samuelgodinho02112020) [reply frontend/package.json:14]: below are the codes one with using @emotion lib and one without 

/** With @emotion/styled **/

import styled from '@emotion/styled';

const Button = styled.button`
  background-color: #007bff;
  color: #ffffff;
  padding: 8px 16px;
  border: none;
  border-radius: 4px;
  cursor: pointer;

  &:hover {
    background-color: #0056b3;
  }
`;

const App = () => {
  return (
    <div>
      <Button>Click me</Button>
    </div>
  );
};

export default App;

without emotion 

/** Without @emotion/styled **/

const buttonStyles = {
  backgroundColor: '#007bff',
  color: '#ffffff',
  padding: '8px 16px',
  border: 'none',
  borderRadius: '4px',
  cursor: 'pointer',
};

const hoverStyles = {
  backgroundColor: '#0056b3',
};

const App = () => {
  return (
    <div>
      <button style={buttonStyles}>Click me</button>
    </div>
  );
};

File: frontend/package.json (hunk 2 above)
Comment (samuelgodinho02112020) [reply frontend/package.json:16]: In simple terms, @types/jest is a package that helps TypeScript understand and work better with Jest, which is a popular testing framework for JavaScript. It provides TypeScript with information about the types and interfaces used in Jest, making it easier to write and maintain tests in TypeScript. Essentially, it improves the development experience when using Jest with TypeScript by adding type safety and better tooling support.

File: frontend/package.json (hunk 1 above)
Comment (castorgodinho) [reply frontend/package.json:14]: Can we remove it?

File: frontend/package.json (hunk 2 above)
Comment (castorgodinho) [reply frontend/package.json:16]: Can we remove it?

**Review by castorgodinho — APPROVED** [review]
LGTM!
